mk/rabbitmq-dist.mk

```diff
@@ -123,23 +163,34 @@ $(DIST_DIR)/%.ez:
 		&& grep -q '^prepare-dist::' $(SRC_DIR)/Makefile) || \
 		$(MAKE) --no-print-directory -C $(SRC_DIR) prepare-dist \
 		APP=$(APP) VSN=$(VSN) EZ_DIR=$(EZ_DIR)
-	$(verbose) (cd $(DIST_DIR) && $(ZIP) $(ZIP_V) -r $*.ez $*)
+	$(verbose) (cd $(DIST_DIR) && $(ZIP) $(ZIP_V) -r `basename $(EZ)` `basename $(EZ) .ez`)
 	$(verbose) rm -rf $(EZ_DIR)
 
+$(MIX_DIST_EZS): $(mix_task_archive_deps_installed)
+	$(verbose) rm -rf $(DIST_DIR)/tmp
+	$(verbose) mkdir $(DIST_DIR)/tmp
+	$(verbose) cd $(SRC_DIR) && MIX do deps.get, deps.compile, compile, archive.build.all -e -o $(realpath $(DIST_DIR))/tmp
+
+$(mix_task_archive_deps_installed):
+	mix archive.install --force https://github.com/hairyhum/mix_task_archive_deps/releases/download/0.1.0/mix_task_archive_deps-0.1.0.ez
+
 # We need to recurse because the top-level make instance is evaluated
 # before dependencies are downloaded.
 
 MAYBE_APPS_LIST = $(if $(shell test -f $(ERLANG_MK_TMP)/apps.log && echo OK),$(ERLANG_MK_TMP)/apps.log)
 
 dist:: $(ERLANG_MK_RECURSIVE_DEPS_LIST) all
+	$(gen_verbose) rm -rf $(DIST_DIR)/*.ez
 	$(gen_verbose) $(MAKE) do-dist DIST_PLUGINS_LIST="$(ERLANG_MK_RECURSIVE_DEPS_LIST) $(MAYBE_APPS_LIST)"
 
 test-dist:: $(ERLANG_MK_RECURSIVE_TEST_DEPS_LIST) test-build
 	$(gen_verbose) $(MAKE) do-dist DIST_PLUGINS_LIST="$(ERLANG_MK_RECURSIVE_TEST_DEPS_LIST) $(MAYBE_APPS_LIST)"
 
 do-dist:: $(DIST_EZS)
-	$(verbose) unwanted='$(filter-out $(DIST_EZS),$(wildcard $(DIST_DIR)/*.ez))'; \
+	$(verbose) unwanted='$(filter-out $(foreach archive,$(wildcard $(DIST_DIR)/*.ez),$(DIST_DIR)/$(notdir $(archive))),$(filter-out $(DIST_EZS),$(wildcard $(DIST_DIR)/*.ez)))'; \
```

Comment:
It makes no sense. There was this "delete all" line in the `dest` target: `$(gen_verbose) rm -rf $(DIST_DIR)/*.ez`, which we probably should keep.
The problem is that MIX dependencies build their dependencies by themselves and it's not easy to get the unwanted list.